docs/agents.md

```diff
@@ -1,128 +1,233 @@
-# Agents
+# Agent
+
+The `BaseAgent` class is the foundation of the Bee Framework, providing the core interface and functionality that all agent implementations must follow. It orchestrates the interaction between LLMs, tools, memory, and development utilities to create intelligent, automated workflows.
+
+## Overview
+
+`BaseAgent` acts as an abstract base class that defines the standard interface and basic functionality for all agents in the framework. It manages the lifecycle of agent operations, coordinates between different components, and provides a consistent interface for agent implementations.
+
+## Architecture
+
+```mermaid
+classDiagram
+    class BaseAgent {
+        +LLM llm
+        +Memory memory
+        +Tool[] tools
+        +DevTools devTools
+        +run(prompt: string, options?: ExecutionOptions)
+    }
+
+    BaseAgent *-- LLM
+    BaseAgent *-- Memory
+    BaseAgent *-- Tool
+    BaseAgent *-- DevTools
+
+    class LLM {
+        +inference()
+        +templates: TemplatePrompt
+    }
+
+    class Memory {
+        +store()
+        +retrieve()
+        +cache: Cache
```

Comment:
There are no such methods.

`BaseMemory` class has the following methods:
- messages: BaseMessage[]
- add(message: BaseMessage): Promise<void>
- delete(message: BaseMessage): Promise<void>
- asReadOnly(): ReadOnlyMemory
- isEmpty(): boolean
- splice(start: number, deleteCount: number, ...items: BaseMessage[])